go/tools/fetch_repo/main.go

```diff
@@ -19,19 +19,45 @@ import (
 )
 
 var (
-	remote = flag.String("remote", "", "Go importpath to the repository fetch")
-	rev    = flag.String("rev", "", "target revision")
-	dest   = flag.String("dest", "", "destination directory")
+	remote     = flag.String("remote", "", "The URI of the remote repository. Must be used with the --vcs flag.")
+	cmd        = flag.String("vcs", "", "Version control system to use to fetch the repository. Should be one of: git,hg,svn,bzr. Must be used with the --remote flag.")
+	rev        = flag.String("rev", "", "target revision")
+	dest       = flag.String("dest", "", "destination directory")
+	importpath = flag.String("importpath", "", "Go importpath to the repository fetch")
 )
 
+func getRepoRoot(remote, cmd, importpath string) (*vcs.RepoRoot, error) {
+	r := &vcs.RepoRoot{
+		VCS:  vcs.ByCmd(cmd),
+		Repo: remote,
+		Root: importpath,
+	}
+
+	if remote != "" && importpath == "" && cmd == "" {
```

Comment:
It's fine to make this an error. I don't think we need to maintain backward compatibility for this tool since it should only be used through the repository rules.

I'll document the change in the next release.